types/digest_of.go

```diff
@@ -0,0 +1,57 @@
+// Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
+//
+// Copyright 2019 Centrifuge GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import "encoding/json"
+
+// DigestOf contains logs
+type DigestOf []DigestItem
+
+// UnmarshalJSON fills u with the JSON encoded byte array given by b
+func (d *DigestOf) UnmarshalJSON(bz []byte) error {
+	var tmp struct {
```

Comment:
Since this struct is defined internally in many places, maybe we can extract it and define it once in the `types` pkg?

Reply:
I would agree in general – DRY. 

At the same time DRY requires naming things, and if they are short (here 3 lines in 4 places) and only serve a helper purpose, I would prefer to keep them local. It's a question of taste, I prioritize dev time and being explicit over succinctness. 

Having said that, I'm happy to change it if you suggest a name :)

Reply:
haha oh naming ... I am fine then, my main concern was if in newer versions the digest of logs content changes we need to update in many places. But not a problem for now.